### PR TITLE
WebSocket publisher plugin

### LINKED ISSUE #105: Create listener programming interface
**Describe the solution you'd like**
It would be good to have a way for external programs (i.e. programs that call the tool as a subprocess) to be able to listen to certain outputs/updates/progressions.

**Additional context**
The worker of the webtool currently does this by reading the console output en watching for keywords, this is less than ideal.

### PR DESCRIPTION
**Pull request by Sne4kers**
Closes #105 
- Added a RabbitMQ plugin that automatically publishes SynTest events 
- Added a WebSocket plugin that automatically publishes SynTest events

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,5 +93,9 @@
       "prettier --config .prettierrc.json --ignore-path .prettierignore --write .",
       "eslint --config .eslintrc.json --ignore-path .eslintignore . --fix"
     ]
+  },
+  "dependencies": {
+    "amqplib": "^0.10.3",
+    "chai-as-promised": "^7.1.1"
```
Comment (mitchellolsthoorn) [inline package.json]: Should these be here? Especially, the amqp dependencies should be removed given that that module is removed entirely.

File: plugins/plugin-core-event-listener-websocket/package.json [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    "@syntest/logging": "*",
+    "@syntest/metric": "^0.1.0-beta.0",
+    "@syntest/module": "*",
+    "ws": "^8.13.0"
+  },
+  "engines": {
+    "node": ">=16"
+  },
+  "devDependencies": {
+    "@types/d3": "^7.4.0",
```
Comment (mitchellolsthoorn) [inline plugins/plugin-core-event-listener-websocket/package.json:62]: Is this dependency used?

File: tools/init/test/example.test.ts [hunk 3]
```diff
@@ -17,16 +17,14 @@
...
-
 const expect = chai.expect;
 
 /**
  * This test is only added such that the github action does not fail.
  */
 describe("example test", () => {
   it("test", () => {
-    getModuleCommand("test");
+    // getModuleCommand("test", <ModuleManager><unknown>{});
```
Comment (mitchellolsthoorn) [inline tools/init/test/example.test.ts:27]: Should this still be fixed

File: package.json (hunk 1 above)
Comment (dstallenberg) [reply package.json]: Already removed it, noticed it too

File: plugins/plugin-core-event-listener-websocket/package.json (hunk 2 above)
Comment (dstallenberg) [reply plugins/plugin-core-event-listener-websocket/package.json:62]: Removed it.

File: tools/init/test/example.test.ts (hunk 3 above)
Comment (dstallenberg) [reply tools/init/test/example.test.ts]: Done.

File: package.json [hunk 4]
```diff
@@ -50,13 +50,15 @@
   "devDependencies": {
     "@commitlint/cli": "17.4.2",
     "@commitlint/config-conventional": "17.4.2",
+    "@types/amqplib": "^0.10.1",
```
Comment (mitchellolsthoorn) [inline package.json:53]: This one can still be deleted

**Review by mitchellolsthoorn — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (dstallenberg) [reply package.json]: Fixed.

**Review by mitchellolsthoorn — APPROVED** [review]

**Pull request merged**